### PR TITLE
[GORDO-1553] Add FatalError conductor state

### PR DESCRIPTION
**Pull request by hkernbach**
### Scope & Purpose

Adds fatal error conductor state.

- [x] :pizza: New feature
- [x] :hammer: Refactoring/simplification

### Checklist

- [x] Tests
  - [x] C++ **Unit tests**

#### Related Information

- [x] GitHub issue / Jira ticket: https://arangodb.atlassian.net/browse/GORDO-1553

### CODE REVIEW

File: arangod/Pregel/Status/ExecutionStatus.h [hunk 1]
```diff
@@ -68,6 +68,7 @@ struct ExecutionTimings {
   Duration storing;
   // FIXME: just sum the times above?
   Duration total;
+  Duration fatal;
```
Comment (jvolmer) [inline arangod/Pregel/Status/ExecutionStatus.h:71]: I don't think we want to measure this duration

File: arangod/Pregel/Conductor/ExecutionStates/FatalErrorState.h [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+#include "Pregel/Worker/Messages.h"
+#include "State.h"
+
+namespace arangodb::pregel::conductor {
+
+struct ConductorState;
+
+struct FatalError : ExecutionState {
+  FatalError(ConductorState& conductor) : conductor{conductor} {
+    conductor._timing.fatal.start();
```
Comment (jvolmer) [inline arangod/Pregel/Conductor/ExecutionStates/FatalErrorState.h:37]: we don't want to start a new timer but want to stop the total timing

Comment (hkernbach) [reply arangod/Pregel/Conductor/ExecutionStates/FatalErrorState.h]: added, thank you

File: arangod/Pregel/Status/ExecutionStatus.h (hunk 1 above)
Comment (hkernbach) [reply arangod/Pregel/Status/ExecutionStatus.h]: Removed.

**Review by jvolmer — APPROVED** [review]

**Pull request merged**